File: 0001-two-sum/0001-two-sum.ts
function twoSum(nums: number[], target: number): number[] {
    const map = {};
    for (let i = 0; i < nums.length; i++) {
        const otherNum = target - nums[i];
        const otherNumIndex = map[otherNum]
        if (otherNumIndex !== undefined) {
            return [otherNumIndex, i];
        } else {
            map[nums[i]] = i;
        }
    };
    return [];
};

File: 0001-two-sum/0001-two-sum.go
func twoSum(nums []int, target int) []int {
    numMap := make(map[int]int)

    for i, num := range nums {
        complement := target - num

        if index, found := numMap[complement]; found {
            return []int{index, i}
        }

        numMap[num] = i
    }

    return nil
}